File: Easy/3392_Count_Subarrays_of_Length_Three_With_a_Condition/Solution.java


public class Solution {
    public int countSubarrays(int[] nums) {
        int n = nums.length;
        int ans = 0;

        for (int i = 2; i < n; i++) {
            double half = nums[i-1] / 2.0;
            double sum = nums[i-2] + nums[i];

            if (sum == half) {
                ans++;
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] nums1 = {1,2,1,4,1};
        System.out.println(solution.countSubarrays(nums1)); // Output: 1
    
        // test cases 2
        int[] nums2 = {1, 1, 1};
        System.out.println(solution.countSubarrays(nums2)); // Output: 0
        
    }
}

File: Easy/3392_Count_Subarrays_of_Length_Three_With_a_Condition/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    int countSubarrays(vector<int>& nums) {
        int n = nums.size();
        int ans = 0;

        for (int i = 2; i < n; i++) {
            float half = nums[i-1] / 2.0;
            float sum = nums[i-2] + nums[i];

            if (sum == half) {
                ans++;
            }
        }

        return ans;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<int> nums1 = {1,2,1,4,1};
    cout << solution.countSubarrays(nums1) << endl;

    // test cases 2
    vector<int> nums2 = {1,1,1};
    cout << solution.countSubarrays(nums2) << endl;

    // [-1,-4,-1,4]

    return 0;
}